File: src/main/java/edu/gbcg/dbInteraction/dbSelector/reddit/comments/CommentSetMapper.java
/*
 * Copyright (c) 2018 Sean Grimes. All rights reserved.
 * License: MIT License
 */

package edu.gbcg.dbInteraction.dbSelector.reddit.comments;

import edu.gbcg.configs.columnsAndKeys.RedditComs;
import edu.gbcg.dbInteraction.dbSelector.RSMapper;

import java.sql.ResultSet;
import java.util.List;
import java.util.Map;

/**
 * Specific RSMapper object for a specific DB type. The implementation for buildMappers is handled in the parent
 * abstract class, however it needs to know the column names for this specific DB so pass it the column names and let
 * it work
 */
public class CommentSetMapper extends RSMapper {
    public CommentSetMapper(Map<String, String> map){ super(map); }
    public CommentSetMapper(){ super(); }
    public List<RSMapper> buildMappers(ResultSet rs){
        return buildMappers_impl(rs, RedditComs.columnNames());
    }
}

File: src/main/kotlin/edu/gbcg/dbInteraction/dbSelector/reddit/comments/CommentSetMapper.kt
/*
 * Copyright (c) 2018 Sean Grimes. All Rights Reserved.
 * License: MIT
 */

package edu.gbcg.dbInteraction.dbSelector.reddit.comments

import edu.gbcg.configs.columnsAndKeys.RedditComs
import edu.gbcg.dbInteraction.dbSelector.RSMapper
import org.json.JSONObject
import java.sql.ResultSet

class CommentSetMapper: RSMapper {
    constructor(map: Map<String, String>): super(map)
    constructor(jsonObject: JSONObject): super(jsonObject)
    constructor(): super()
    override fun buildMappers(rs: ResultSet): MutableList<RSMapper>? {
        return buildMappersImpl(rs, RedditComs.columnNames())
    }
}
